Apps/WebClient/src/NewClientApp/src/components/navmenu/HeaderComponent.vue

```diff
@@ -0,0 +1,391 @@
+    navbarStore.setHeaderState(isOpen);
+}
+
+function onScroll(): void {
+    const st = window.scrollY || document.documentElement.scrollTop;
+    if (
+        Math.abs(st - lastScrollTop.value) > minimumScrollChange &&
+        isMobileWidth.value
+    ) {
+        if (st > lastScrollTop.value) {
+            // down-scroll
+            if (isHeaderShown.value) {
+                setHeaderState(false);
+            }
+        } else {
+            // up-scroll
+            if (!isHeaderShown.value) {
+                setHeaderState(true);
+            }
+        }
+    }
+    // For Mobile or negative scrolling
+    lastScrollTop.value = st <= 0 ? 0 : st;
+}
+
+function handleToggleClick(): void {
+    toggleSidebar();
+}
+
+function handleLogoutClick(): void {
+    if (isValidIdentityProvider.value) {
+        showRating();
+    } else {
+        processLogout();
+    }
+}
+
+function handleShowTourClick(): void {
+    hasViewedTour.value = true;
+    appTourComponent.value?.showModal();
+}
+
+function showRating(): void {
+    ratingComponent.value?.showModal();
+}
+
+function processLogout(): void {
+    logger.debug(`redirecting to logout view ...`);
+    router.push({ path: "/logout" });
+}
+
+watch(isMobileWidth, (value) => {
+    if (!value) {
+        setHeaderState(false);
+    }
+});
+
+watch(
+    () => route.query,
+    (value) => {
+        if (value.registration === "success") {
+            router.replace({ query: {} });
+            appTourComponent.value?.showModal();
+        }
+    }
+);
+
+onUnmounted(() => {
+    window.removeEventListener("scroll", onScroll);
+});
+
+nextTick(() => {
+    window.addEventListener("scroll", onScroll);
+    if (!isMobileWidth.value) {
+        setHeaderState(false);
+    }
+});
+</script>
+
+<template>
+    <v-app-bar
+        :scroll-behavior="!isHeaderShown ? 'hide' : undefined"
+        class="border-b-md border-accent border-opacity-100"
+        color="primary"
+    >
+        <template #prepend v-if="isSidebarButtonShown">
+            <HgIconButtonComponent
+                :icon="isSidebarOpen ? 'fas fa-times' : 'fas fa-bars'"
+                @click="handleToggleClick"
+            />
+        </template>
+        <v-img
+            alt="Go to health Gateway home page"
+            to="/"
+            src="@/assets/images/gov/hg-logo-rev.svg"
+            max-width="143px"
+            class="pa-1"
+        />
+        <v-spacer />
+        <HgIconButtonComponent
+            v-if="isAppTourAvailable"
+            @click="handleShowTourClick"
+            data-testid="app-tour-button"
+        >
+            <v-badge color="red" :value="highlightTourChangeIndicator">
+                <v-icon icon="fas fa-lightbulb" />
+            </v-badge>
+        </HgIconButtonComponent>
+        <HgIconButtonComponent
+            v-if="isNotificationCentreAvailable"
+            @click="notificationButtonClicked = true"
+            data-testid="notification-centre-button"
+        >
```

Comment:
This will need to be wired up to toggle the notification centre when that is implemented.